### PR TITLE
fix: fix type definition for sync payload

### PR DESCRIPTION
**Pull request by Khaledgarbaya**

### CODE REVIEW

File: index.d.ts [hunk 1]
```diff
@@ -95,7 +95,7 @@ export interface Space {
 export interface SyncCollection {
     entries: Array<Entry<any>>;
     assets: Array<Asset>;
-    deletedEntries: Array<Entry<any>>;
+    deletedEntries: Array<any>;
     deletedAssets: Array<Asset>;
```
Comment (zcei) [inline index.d.ts]: The same should be for `deletedAssets` - they won't resolve with the whole Asset, but only a meta-description

**Comment by googol** [comment]
If deletedEntries and deletedAssets don't contain Entries/Assets, then what do they contain? Can that be modelled in a new interface?

The any type is likely too broad

**Comment by zcei** [comment]
@googol yep, that's way better. It should be around this:

```ts
interface SpaceLink {
  id: string
  type: 'Link'
  linkType: 'Space'
}

enum SyncType {
  DeletedEntry = 'DeletedEntry',
  DeletedAsset = 'DeletedAsset'
}

interface SyncDeletedEntity {
  sys: {
    space: SpaceLink
    id: string
    type: SyncType
    createdAt: string
    updatedAt: string
    deletedAt: string
    revision: number
  }
}
```

I just figured, that the `Sys` interface always has a property `contentType`, which doesn't match all cases. Maybe this should be turned into some kind of generic, or have multiple different `Sys` interfaces? Same for Links, we currently have `ContentTypeLink` and this would now introduce a `SpaceLink`

**Comment by Khaledgarbaya** [comment]
Alright I will change that asap

**Comment by googol** [comment]
Different Sys interfaces would be ideal. Let me know if I can help!

File: index.d.ts [hunk 2]
```diff
@@ -95,7 +95,7 @@ export interface Space {
...
     nextSyncToken: string;
     toPlainObject(): SyncCollection;
```
Comment (grub203) [inline index.d.ts]: 7248068f53cf331d6dca4cd6f4d057926dd6483f

**Comment by makinwab** [comment]
Closing this PR as it's been open for a couple of years now. If there's a very much needed change we need to incorporate, please feel free to reopen.

**Pull request closed without merging**